### ISSUE TITLE
Tautulli - multiple server support?

### ISSUE THREAD

**Issue by n1nj4888** [DirtyCajunRice/grafana-scripts#25]
Can the Tautulli script be modified to support multiple tautulli servers (radarr/sonarr scripts already seem to support multiple servers)? 

This is especially useful for those with more than one Plex server since I understand Tautulli currently is only able to monitor a single Plex server, therefore 2 Tautulli instances are required to monitor 2 Plex servers?

**Comment by samwiseg00** [comment]
This is not a simple task. Since we use the session_key as the unique ID there can be cases where the same key could be present in both servers. 

Have you tried running multiple instances of the `tautulli.py` script? Theoretically that should work.

**Comment by n1nj4888** [comment]
By session_key, are you referring to session_id (or session_id1) that is stored in the influxDB Tautulli mearement? This appears to be a 24 character alpha-numeric ID which is unlikely to be the same across servers but, if it is, could possibly a unique key be stored in addition to the current session_id / session_id1 keys which is something like <SERVER_URL>:<SERVER_PORT>_<SESSION_ID>?

Thanks!

**Comment by n1nj4888** [comment]
Thanks @samwiseg00, understood...

 could try running a separate docker for Server #2 to write to either the same DB/measurement (with the possibility that this may break the Session Key) or possibly get the docker for Server #2 to write to a separate DB but this would be better if it could write to the same DB/measurement as Server #1 so that the Grafana Dashboard could report on both servers...

Is the Session Key truly unique to the DB/measurement table (like a  primary key) or could we simply modify the script t also write the Server URL:Port as a new column and distinguish them in the dashboard?

I understand this may require some work, so leaving it here as a suggestion for the future if you agree it would be beneficial to support multiple tautulli servers (similar to how it supports multiple sonarr/radarr servers currently)...

**Comment by samwiseg00** [comment]
We will consider it. But it will not happen any time in the near future. 

Note you will have to use a single database as grafana will not pull from 2 sources in one panel. You can duplicate the current panel and point it to the other database if you would like to go that route. 

If we do it we would make it a unified view and import into the same database. That is why is is much more complicated then you might think ;)

**Comment by samwiseg00** [comment]
The short term solution is what I described. 2 instances pushing to different databases and you can duplicate the panels to show the other servers. _If_ any development is done it will be a complete solution and not something halfed baked.

**Comment by samwiseg00** [comment]
@n1nj4888 After some thought and discussion with the group. We will not be pursuing development of this feature. As described there are many complications with keeping everything in one script and the work can be acomplished ont he Grafana side. 

I did however do some testing and realized that it is pretty easy to get things into a unified view. I had forgotten that Grafana has a `--mixed--` option to mix the different data sources. 

The best/scaleable way that I tested is to have a script instance per Tautulli server and posting each server into a separate database.

![Table](https://i.imgur.com/xVUrWgD.png)
